stl/inc/utility

```diff
@@ -644,6 +645,77 @@ template <size_t _Idx>
 inline constexpr in_place_index_t<_Idx> in_place_index{};
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+
+template <typename _Ty>
+inline constexpr bool _Is_integral_not_bool_char = is_integral_v<_Ty>
+    && !is_same_v<_Ty, bool> && !is_same_v<_Ty, char>
+    && !is_same_v<_Ty, char8_t> && !is_same_v<_Ty, char16_t>
+    && !is_same_v<_Ty, char32_t> && !is_same_v<_Ty, wchar_t>;

```

Comment:
We can shorten this to:
```suggestion
template <class _Ty>
inline constexpr bool _Is_integral_not_bool_char = is_integral_v<_Ty>
    && !_Is_any_of_v<_Ty, bool, char, 
#ifdef __cpp_char8_t
        char8_t, 
#endif // __cpp_char8_t
        char16_t, char32_t, wchar_t>;
```

Reply:
Counts the number of lines o.O

But even if its longer It seems more correct given that `char8_t `might not be defined

Reply:
I think something like the following would be best for the hyper-throughput-paranoid, that's how we did _Is_character in <xutility>:

```
template<class _Ty>
inline constexpr bool _Is_integral_non_bool_char = is_integral_v<_Ty>;
template<>
inline constexpr bool _Is_integral_non_bool_char<char> = false;
template<>
inline constexpr bool _Is_integral_non_bool_char<char16_t> = false;
// repeat for each
```

Reply:
Should concepts be used here? @CaseyCarter